cartography/intel/aws/ec2.py

```diff
@@ -432,6 +472,129 @@ def load_ec2_auto_scaling_groups(session, data, region, current_aws_account_id,
                 )
 
 
+def load_load_balancer_v2s(session, data, region, current_aws_account_id, aws_update_tag):
+    """
+    For each elbv2
+        'LoadBalancers': [
+            {
+                'LoadBalancerArn': 'string',
+                'DNSName': 'string',
+                'CanonicalHostedZoneId': 'string',
+                'CreatedTime': datetime(2015, 1, 1),
+                'LoadBalancerName': 'string',
+                'Scheme': 'internet-facing''internal',
+                'VpcId': 'string',
+                'State': {
+                    'Code': 'active''provisioning''active_impaired''failed',
+                    'Reason': 'string'
+                },
+                'Type': 'application''network',
+                'AvailabilityZones': [
+                    {
+                        'ZoneName': 'string',
+                        'SubnetId': 'string',
+                        'LoadBalancerAddresses': [
+                            {
+                                'IpAddress': 'string',
+                                'AllocationId': 'string'
+                            },
+                        ]
+                    },
+                ],
+                'SecurityGroups': [
+                    'string',
+                ],
+                'IpAddressType': 'ipv4''dualstack'
+                'Listeners': [
+                    {
+                        'ListenerArn': 'string',
+                        'LoadBalancerArn': 'string',
+                        'Port': 123,
+                        'Protocol': 'HTTP'|'HTTPS'|'TCP'|'TLS'|'UDP'|'TCP_UDP',
+                        'Certificates': [
+                            {
+                                'CertificateArn': 'string',
+                                'IsDefault': True|False
+                            },
+                        ],
+                        'SslPolicy': 'string',
+                        'DefaultActions': [
+                            {
+                                "TargetGroupArn": "arn:aws:elasticloadbalancing:us-east-1:XXXXXXXXXXXX:targetgroup/Y",
+                                "Type": "forward"
+                            }
+                        ]
+                    }
+                ],
+                'TargetGroups': [
+                    'TargetGroupArn': 'string',
+                    ...
+                    'TargetType': 'instance'|'ip'|'lambda',
+                    'Targets': ["i-0f76fade"]
+                ]
+            },
+        ],
+    """
+    ingest_load_balancer_v2 = """
+    MERGE (elbv2:LoadBalancerV2{id: {ID}})
```

Comment:
Could we leverage the LoadBalancer node and add a version attribute? This would help build queries to look at all load balancer in a simpler query. Another trick could be to use label/node "inheritence" and use

MERGE (elbv2:LoadBalancer:LoadBalancerV2{id: {ID}})

Reply:
Hey @sachafaust,

I've been thinking about it and here are the potential rough spots:

1) Classic ELBs have both SOURCE SECURITY GROUPS and SECURITY GROUPS.  The distinction is important for identifying access through a classic ELB.  ALB's have attached security groups, but there's no concept of a source security group.  So any graphql query which relies on `(lb:LoadBalancer)-[SOURCE_SECURITY_GROUP]->(sg:EC2SecurityGroup)` would fail to return the appropriate results on an ALB.

2) This one is probably bigger than #1.  The next PR that would build on this one would improve our visibility around NLBs.  NLB's are also a type of `elbv2`, but with type `network` instead of type `application`.  NLB's don't have any security groups directly attached.  They may have security groups attached to ENI's.  NLBs can be included in AWS PrivateLink, which itself has a kind of resource policy which can allow endpoint connections from different IAM Users, Roles, or AWS accounts. So the queries that will be required to identify NLB's that might be internet accessible, or connected to an internet-connectable privatelink, will be very different from both classic ELBs and ALBs. This makes me think we should treat them as different technologies in GraphQL so we can avoid overly complex edge cases on our queries. (There may be more to learn about NLBs; I'm learning now.)

3) If we treat classic ELB, ALB, and NLB the same, then would we also treat an APIGateway the same?  At some point our GraphQL representation of our AWS environment may start to be difficult to understand because it's oversimplifying the environment. Looking at how the tool has been built out until now, it looks like it's trying to built an accurate model of AWS, leaving simplifications to things like linking `EC2Instances` directly to `LoadBalancers` instead of more accurately connecting them through `ELBListeners`.  I followed this kind of simplification for ALBs. For ALBs the accurate representation would be `(LoadBalancerV2)-->(ELBV2Listener)-->(TargetGroup)-->(EC2Instance)`.

So with all of that information, could I ask you to think about how we should be building this out, not just for ALBs, but also for NLBs and APIGateways, and let me know which direction I should take? Thanks a bunch for this tool.

Unrelated: As the author of `policyuniverse`, I'll try and jump on those issues at some point :).

Reply:
@sachafaust ^ Waiting for your input before we proceed.

Reply:
@sachafaust - Should I continue to wait for your thoughts your just proceed with your original guidance?

Reply:
I think it make sense to proceed but this goes beyond my understanding of AWS ELB.  Looping the eng folks on our end. @ecdavis @achantavy I'll defer to you.

Reply:
I think it's worth keeping them separate for the reasons listed above. Thanks for the in-depth explanation @scriptsrc -- aiming to get this merged soon.